### PR TITLE
Update action and tests for the Eigenvalue executioner

### LINKED ISSUE #177: Add support for `Eigenvalue` eigensolver
This issue can be closed when `NtAction` supports using the `Eigenvalue` executioner and new tests are added for criticality search using `Eigenvalue`.

Our existing framework for solving criticality search problems involve using the `NonlinearEigen` or `InversePowerMethod` executioners. These executioners require `EigenKernels` for the neutron source terms, namely the `CoupledFissionEigenKernel` and `DelayedNeutronEigenSource`.

`NonlinearEigen` and `InversePowerMethod` still exist, but are now deprecated in favor of the new `Eigenvalue` executioner. However, the `Eigenvalue` executioner requires:
1. SLEPc and PETSc version > 3.13 for optimal solve convergence performance (The cray-optimized PETSc on Blue Waters is only 3.9.3.0)
2. regular `Kernels` for the neutron source terms with `extra_vector_tags = 'eigen'`, which means we have to edit our automated initializer `NtAction`.

While it's technically possible to use the `Eigenvalue` executioner in Moltres right now, it would be a tedious process of manually writing the variable, kernel, BC, etc. objects for all neutron groups because `NtAction` does not support the right format for `Eigenvalue`.

We can't completely remove support for the deprecated `Executioners` at this moment because `cray-petsc` on Blue Waters is only at 3.9.3.0 and we need Blue Waters for large simulations which need more memory/computing power.

### LINKED ISSUE #197: Remove deprecated MOOSE code
CIVET has flagged Moltres as having deprecated code: https://civet.inl.gov/job/1027414/

Notably, the following error is seen on most of the failing tests:
```
Deprecated code:
'InversePowerMethod' executioner is deprecated in favor of 'Eigenvalue' executioner.
Few parameters such as 'bx_norm', 'k0', 'xdiff', 'max_power_iterations', 'min_power_iterations', 'eig_check_tol', 'sol_check_tol', and 'output_before_normalization' are no longer supported.
However, 'Eigenvalue' executioner supports more solving options by interfacing SLEPc.
```

And the following is seen in the `laminar_flow.boussinesqSquare` test:
```
Deprecated code:
The parameter p is deprecated.
The coupled variable parameter 'p' has been deprecated and will be removed 1/1/2022. Please use the 'pressure' coupled variable parameter instead.
```

This issue can be closed when:

- [x] The deprecated code outlined in the test results linked above have been addressed and the "Test deprecated" target completes successfully. 

Moltres members with sufficient permissions should be able to turn on the "Test deprecated" target manually on the PR testing screen to ensure this condition is satisfied. Feel free to reach out if you have any questions!

### PR DESCRIPTION
**Pull request by smpark7**
## Description
Closes #177 
This PR transitions Moltres from the old `NonlinearEigen`/`InversePowerMethod` executioner implementation to the new `Eigenvalue` executioner implementation in MOOSE for solving k-eigenvalue problems. The old implementation is deprecated and will eventually be removed from MOOSE. All relevant tests and documentation are updated to reflect the transition to the new implementation.

A k-eigenvalue problem may be formulated as $Lx = \frac{1}{k} Fx$, where the $F$ operator comprises of neutron source terms and the $L$ operator comprises of the streaming, collision, and scattering terms.

Unlike the old implementation which requires a separate `EigenKernel` class for the $F$ operator, the new implementation uses the regular `Kernel` class with the `extra_vector_tags` parameter to indicate whether it belongs in $F$ (otherwise $L$).

This PR also refactors `NtAction` to reduce code duplication. `NtAction` is an Action object which streamlines the process of creating Moltres input file by automatically setting up various Kernels, BCs, etc. required in a neutronics simulation.

## Changes
- Remove deprecated EigenKernels in `src/` & `include/`, i.e., `CoupledFissionEigenKernel`, `PrecursorEigenSource`, `SelfFissionEigenKernel`
- Remove documentation files in `doc/` associated with the EigenKernels
- Update `NtAction` and `PrecursorAction` to use the new `EigenProblem` implementation
- Update test input and output files in `tests/` to use the new `EigenProblem` implementation
- Update input files in `problems/` to use the new `EigenProblem` implementation
- Update the instructions on how to run the `2021-cnrs-benchmark` input files and the expected computational requirements
- Update tutorial input files in `tutorials/` to use the new `EigenProblem` implementation
- Update the input syntax and eigenvalue tutorials to use the new `EigenProblem` implementation
- Fix wrong int signedness in `ElmIntegFissPostprocessor` and `ElmIntegFissNtsPostprocessor` in `src/` and `include/`
- Refactor `NtAction` to reduce code duplication

## Impact
- Brings Moltres up to date with the new `Eigenvalue` executioner implementation in MOOSE
- Improved code cleanliness in `NtAction`

### CODE REVIEW

File: doc/content/getting_started/input_syntax.md [hunk 1]
```diff
@@ -601,8 +608,9 @@ values during the simulation, which may be desired when running test runs.
 
 ## Debug Block id=debug
 
-This simply tells Moltres to print the variable residual norms during the
-non-linear solve.
+This parameter simply tells Moltres to print the variable residual norms during the
```
Comment (yardasol) [inline doc/content/getting_started/input_syntax.md:611]:
```suggestion
This parameter tells Moltres to print the variable residual norms during the
```
Not needed, but feel free to ignore it feels like I'm erasing your voice from the docs.

File: include/postprocessors/ElmIntegTotFissNtsPostprocessor.h [hunk 2]
```diff
@@ -4,7 +4,12 @@
 #include "MooseVariableInterface.h"
 
 class ElmIntegTotFissNtsPostprocessor : public ElementIntegralPostprocessor
-/* public MooseVariableInterface */
+/**
+ * This class computes the postprocessor value for the total number of
+ * neutrons produced in one neutron generation from fission and delayed
+ * neutron precursors if specified.
+ * \f$ \sum_g \nu \Sigma_{g,f} \phi_g + \sum_i \lambda_i C_i \f$
```
Comment (yardasol) [inline include/postprocessors/ElmIntegTotFissNtsPostprocessor.h:11]: Is it possible to rename this class? That name is very hard to read and understand.

File: problems/2021-cnrs-benchmark/phase-1/buoyancy.i [hunk 3]
```diff
@@ -302,33 +289,33 @@ t_alpha = 2e-4 # K-1, Thermal expansion coefficient
   [ntsApp]
     type = FullSolveMultiApp
     app_type = MoltresApp
-    execute_on = timestep_begin
+    execute_on = timestep_end
```
Comment (yardasol) [inline problems/2021-cnrs-benchmark/phase-1/buoyancy.i:292]: Why this change?

File: src/actions/NtAction.C [hunk 4]
```diff
@@ -395,3 +261,72 @@ NtAction::act()
...
+    params.set<std::vector<SubdomainName>>("block") =
+        getParam<std::vector<SubdomainName>>("pre_blocks");
+  if (isParamValid("use_exp_form"))
+    params.set<bool>("use_exp_form") = getParam<bool>("use_exp_form");
+  std::vector<std::string> include = {"temperature", "pre_concs"};
+  params.applySpecificParameters(parameters(), include);
+  params.set<unsigned int>("num_precursor_groups") = _num_precursor_groups;
+  std::string kernel_name = "DelayedNeutronSource_" + var_name;
+  _problem->addKernel("DelayedNeutronSource", kernel_name, params);
+}
```
Comment (yardasol) [inline src/actions/NtAction.C:332]: Good refactor!

**Review by yardasol — APPROVED** [review]
Hi @smpark7,

Good work on this code update and refactor. You did a good job updating the docpages. I'm going to approve, but am curious why many `int` variables were changed to `unsigned int`.

File: include/postprocessors/ElmIntegTotFissNtsPostprocessor.h (hunk 2 above)
Comment (smpark7) [reply include/postprocessors/ElmIntegTotFissNtsPostprocessor.h:11]: Good point. I created issue #263 to track this. I won't rename it here because many input files use this postprocessor and editing all of them would bloat up this PR.

File: problems/2021-cnrs-benchmark/phase-1/buoyancy.i (hunk 3 above)
Comment (smpark7) [reply problems/2021-cnrs-benchmark/phase-1/buoyancy.i:292]: This makes the thermal-hydraulics problem run before the neutronics problem. I found that this order reduces simulation time. The final result is unchanged because this simulation terminates at steady state.

File: src/actions/NtAction.C (hunk 4 above)
Comment (smpark7) [reply src/actions/NtAction.C:332]: Thanks! :)

**Comment by smpark7** [comment]
Thanks for the review @yardasol! I changed the parameters to unsigned int for consistency. All other occurrences of those parameters in Moltres are unsigned ints. Feel free to merge if you (re)approve.

**Review by yardasol — APPROVED** [review]
Good work @smpark7!
